### PR TITLE
8266643: Intermittent failure of HonorDeveloperSettingsTest unit test

### PR DESCRIPTION
**Pull request by arapte**
The test fails intermittently on github build. But I do not observe this issue locally on my Ubuntu20.04 VM.
I am suspecting that the reason of failure is that the test are not cleaned up correctly. The changes in this PR cleans up the test, by removing all children added to root and hiding the stage.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266643](https://bugs.openjdk.java.net/browse/JDK-8266643): Intermittent failure of HonorDeveloperSettingsTest unit test

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/496/head:pull/496` \
`$ git checkout pull/496`

Update a local copy of the PR: \
`$ git checkout pull/496` \
`$ git pull https://git.openjdk.java.net/jfx pull/496/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 496`

View PR using the GUI difftool: \
`$ git pr show -t 496`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/496.diff">https://git.openjdk.java.net/jfx/pull/496.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.graphics/src/test/java/test/javafx/css/HonorDeveloperSettingsTest.java [hunk 1]
```diff
@@ -70,6 +72,8 @@ private static void resetStyleManager() {
     @After
     public void cleanup() {
         resetStyleManager();
+        root.getChildren().clear();
+        stage.hide();
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/test/java/test/javafx/css/HonorDeveloperSettingsTest.java:76]: I recommend protecting these with null checks.

File: modules/javafx.graphics/src/test/java/test/javafx/css/HonorDeveloperSettingsTest.java [hunk 2]
```diff
@@ -57,6 +57,8 @@
     private Scene scene;
     private Rectangle rect;
     private Text text;
+    Stage stage;
+    Group root;
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/test/java/test/javafx/css/HonorDeveloperSettingsTest.java:61]: Can these be private?

**Comment by arapte** [comment]
Hello Kevin,
Adding a call to `StyleManager.getInstance().forget(scene)` seems like a proper fix because it clears the `StyleManager`.
This PR executed the tests two times using only this call and did not fail the test. 
Does this look good OR 
should we also include `root.getChildren().clear()` and `stage.hide()` calls ?

**Comment by kevinrushforth** [comment]
> should we also include `root.getChildren().clear()` and `stage.hide()` calls ?

I see no need for `root.getChildren().clear()`. I think all graphics unit tests that call `stage.show()` should call `stage.hide()` in a cleanup method, but I see that most of them don't. Maybe that can be done globally in a follow-up test bug (perhaps in a future test sprint)?

**Review by kevinrushforth — APPROVED** [review]
Looks good. In addition to testing this locally, I kicked off two GitHub actions builds (the Linux systems used by GitHub actions is are the only systems on which we've seen this fail), and both passed.

**Comment by arapte** [comment]
> Maybe that can be done globally in a follow-up test bug (perhaps in a future test sprint)?

Sounds good to me too, shall raise a bug to track.

**Pull request closed without merging**